### PR TITLE
feat(jwt-auth): support configuring `key_claim_name`

### PR DESCRIPTION
**Pull request by shreemaan-abhishek**
### Description

Support configuring key_claim_name like kong: https://docs.konghq.com/hub/kong-inc/jwt/configuration/#config-key_claim_name

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Review by Revolyssup — APPROVED** [review]

**Review by membphis — COMMENTED** [review]
need to update the doc at same time

**Review by Revolyssup — APPROVED** [review]

**Review by nic-6443 — APPROVED** [review]

File: apisix/plugins/jwt-auth.lua [hunk 1]
```diff
@@ -48,7 +48,11 @@ local schema = {
         hide_credentials = {
             type = "boolean",
             default = false
-        }
+        },
+        key_claim_name = {
+            type = "string",
```
Comment (membphis) [inline apisix/plugins/jwt-auth.lua:53]: Add a limitation about the length, the minimum length should be 1

empty string isn’t acceptable

File: apisix/plugins/jwt-auth.lua [hunk 2]
```diff
@@ -48,7 +48,12 @@ local schema = {
...
+            default = "key",
+            minLength = "1",
```
Comment (membphis) [inline apisix/plugins/jwt-auth.lua:55]: this wrong, we should use `minLength = 1`, it should be a number

https://json-schema.org/understanding-json-schema/reference/string

image

Comment (shreemaan-abhishek) [reply apisix/plugins/jwt-auth.lua]: fixed

**Review by nic-6443 — APPROVED** [review]

**Comment by mikyll** [comment]
This PR fixed #11276

**Pull request merged**